ui/src/app/services/workspace-storage.service.ts

```diff
@@ -40,15 +47,17 @@ export class WorkspaceStorageService {
         }
         throw e;
       });
-    this.cache.set(key, load);
-    return load;
+    nextCurrentWorkspace(workspace);
```

Comment:
I don't think I agree with the idea that interactions with `WorkspaceStorageService` should affect the currentWorkspaceStore. Those interactions are currently handled by the workspace-wrapper, which is not perfect, but it is at least centralized to a clear owner and matches what is rendered on the page (there is nothing to indicate that if I call `WorkspaceStorageService.getWorkspace()`, I'd like to immediately update my context and change things currently shown on the page).

I would instead suggest perhaps plumbing a `reloadWorkspace()` callback down from the workspace-wrapper to its children, or having workspace-wrapper listen for updates on the workspace store and perhaps even having the workspace edit page trigger the update with the reloaded value, e.g.:

```
reloadWorkspace().then((ws) => currentWorkspaceStore.next(ws))
```

Reply:
That seems reasonable to me. I think if workspaceWrapper is subscribed to the store, we may just get it for free. I will try it

Reply:
It broke when I removed these from the storage service. If you like, we can hop on a call to look into it? I'm trying to avoid passing a reload down to every child component.

Reply:
How about my second suggestion?

> or having workspace-wrapper listen for updates on the workspace store and perhaps even having the workspace edit page trigger the update with the reloaded value

Reply:
i.e. workspace-edit page calling `currentWorkspaceStore.next`

Reply:
Hmmm... let me try that. `currentWorkspaceStore` uses WorkspaceData, so there is a bit of manipulation/merging of data structures, but it could work. However, from a fundamental perspective, what is the purpose of `WorkspaceStorageService` if it isn't affecting `currentWorkspaceStore`. My understanding was that `currentWorkspaceStore` just exposes the current workspace in the cache.

Reply:
I'm worried we are just generating two workspace caches, and should just delete `WorkspaceStorageService` if so.